File: execs/c++/11.cpp
#include<iostream>
#include<cmath>

bool isPrime(int n)
{
    if(n==1)
        return 0;
    else if(n==2)
        return 1;
    else
    {
            for(int i=2;i<sqrt(n)+18;i++)
            {
                    if(n%i==0)
                    {
                        return 0;
                    }
            }

    }
    return 1;
}

int main()
{
    int n;
    std::cout<<"Enter a number: ";
    std::cin>>n;
    if(isPrime(n))
        std::cout<<n<<" is prime.";
    else
        std::cout<<n<<" is not prime.";

    return 0;
}

File: execs/perl/11.pl
#!/usr/bin/perl

use strict;
use warnings;
use Math::Complex;

sub isPrime
{
	my $num=$_[0];
	my $prime=1;
	
	if($num==1)
	{	
		$prime=0;
	}	
	elsif($num==2)
	{
		$prime=1;
	}
	else
	{
		for(my $i=2;$i<sqrt($num)+1;$i+=1)
		{
			if($num%$i==0)
			{	
				$prime=0;
				last;
			}			
		}
	}
	return $prime;
}

print("Enter a number: "); 
my $number=<STDIN>;

if(isPrime($number))
{
	printf("%d is prime\n",$number);
}
else
{
	printf("%d is not prime\n",$number);
}




